lua/neogit.lua

```diff
@@ -39,8 +40,7 @@ local setup = function(opts)
   hl.setup()
   signs.setup()
   state.setup()
-
-  require("neogit.autocmds").setup()
+  watcher.setup()
```

Comment:
I don't think the watcher should start inside config.setup, but rather when you open the status buffer.

How does this handle `Neogit cwd="other"` or when multiple different repos are used.

Additionally, I don't see any cleanup, as with file watchers, it is very common to create `EMFILE: Too many open files`, especially if running multiple watchers. The number of plugins that I've encountered this exact error for when they are using fs watching is quite high, and I've also run into this error with syncthing as well, though it catches it and handles it by reducing the watching.

Reply:
> I don't think the watcher should start inside config.setup, but rather when you open the status buffer.
Reasonable. It's to trigger a refresh of the `status` buffer, so, the init method of the `status` module seems like a good spot.

> How does this handle `Neogit cwd="other"` or when multiple different repos are used.
It.. doesn't. In fact, I think my previous refactor of extracting `status.repo` into `git.repo` and treating it like a singleton broke that. I'll try to work out a way to support this.


> Additionally, I don't see any cleanup, as with file watchers, it is very common to create `EMFILE: Too many open files`, especially if running multiple watchers. The number of plugins that I've encountered this exact error for when they are using fs watching is quite high, and I've also run into this error with syncthing as well, though it catches it and handles it by reducing the watching.

Fair. Since this isn't recursive, it's just watching the top-level of `./.git/` for changes, so there are only a handful of files watched. Still, it would be easy to use `:stop()` when the status buffer closes.